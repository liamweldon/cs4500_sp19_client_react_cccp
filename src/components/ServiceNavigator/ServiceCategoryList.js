import React from 'react'

const ServiceCategoryList = categories => {
   console.log("CATEGORIES");
   console.log(categories);
   return (<ul className="list-group">
        {
            categories.map(serviceCategory =>
                <li key={serviceCategory.id}
                    className="list-group-item no-border">
                    <a href={`#${serviceCategory.id}`}>{serviceCategory.title}</a>
                </li>
            )
        }
    </ul>)
}
    

export default ServiceCategoryList