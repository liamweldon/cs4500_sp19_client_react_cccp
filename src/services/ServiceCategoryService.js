export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if(ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }
    findServiceCategoryById = categoryId =>
        fetch(`http://localhost:8080/api/service-categories/${categoryId}`)
            .then(response => response.json())
    findAllServiceCategories = () =>
        fetch("http://localhost:8080/api/service-categories")
            .then(response => response.json())
}
