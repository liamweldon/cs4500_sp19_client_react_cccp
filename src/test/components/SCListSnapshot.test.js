import React from "react";
import TestRenderer from "react-test-renderer";

import ServiceCategoryList from "../../components/ServiceNavigator/ServiceCategoryList";
import ServiceCategoryService from "../../services/ServiceCategoryService";
import mockService from "../services/mocks/MockServiceCategoryService";
import serviceCategories from '../mockData/service-categories.mock.json'

const serviceCategoryService = ServiceCategoryService.getInstance();
beforeEach(() => {
  global.fetch = mockService(serviceCategories);
});

var test_new_category = {
	id: "999",
	serviceCategoryName: "New One",
	serviceProviders: [],
  questions: [],
  services: [] 
}

// Snapshot testing code
test("Rerender works properly", () => {
  // Create snapshot of initial services
  return serviceCategoryService
    .findAllServiceCategories()
    .then(categories => {
      let component = TestRenderer.create(
        <ServiceCategoryList categories={categories} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      // Add a new service category to the list, and re-render. Since there's 
      // No real interactivity changing the dom in this component, this is the only
      // real way to test rendering changes.
      categories.push(test_new_category);

      component = TestRenderer.create(
            <ServiceCategoryList categories={categories} />);
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })
    .catch(function(error) {});
});