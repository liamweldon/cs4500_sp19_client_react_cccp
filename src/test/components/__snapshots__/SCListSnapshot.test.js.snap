// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rerender works properly 1`] = `
<div
  style={
    Object {
      "position": "fixed",
    }
  }
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item no-border"
    >
      <a
        href="#123"
      >
        Home Improvements
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#345"
      >
        Pets
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#234"
      >
        Wellness
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#456"
      >
        Business
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#567"
      >
        Events
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#678"
      >
        Crafts
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#789"
      >
        Design and Web
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#890"
      >
        Legal
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#987"
      >
        Personal
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#876"
      >
        Photography
      </a>
    </li>
  </ul>
</div>
`;

exports[`Rerender works properly 2`] = `
<div
  style={
    Object {
      "position": "fixed",
    }
  }
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item no-border"
    >
      <a
        href="#123"
      >
        Home Improvements
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#345"
      >
        Pets
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#234"
      >
        Wellness
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#456"
      >
        Business
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#567"
      >
        Events
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#678"
      >
        Crafts
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#789"
      >
        Design and Web
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#890"
      >
        Legal
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#987"
      >
        Personal
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#876"
      >
        Photography
      </a>
    </li>
    <li
      className="list-group-item no-border"
    >
      <a
        href="#999"
      >
        New One
      </a>
    </li>
  </ul>
</div>
`;
